tests/dismod_execution/dismod_tests.py

```diff
@@ -0,0 +1,196 @@
+from cascade_at.dismod.api.dismod_io import DismodIO
+from cascade_at.dismod.api.run_dismod import run_dismod
+
+tol_fixed = 1e-10
+tol_random = 1e-10
+
+if 0:
+    def system(args):
+        cmd = ' '.join(args)
+        rtn = os.system(cmd)
+        if rtn != 0:
+            raise Exception (f"ERROR -- return status {rtn} from: '{cmd}' ")
+        return rtn
+else:
+    def system(args):
+        # run_dismod assumes that the execitable name is dmdismod
+        # To make this code work with the dismod_at executable, make an executable simlink: dmdismod -> dismod_at
+        program = args[0]
+        db = args[1]
+        cmd = ' '.join(args[2:])
+        info = run_dismod(db, cmd)
+        if info.stdout or info.stderr: print ('-'*30)
+        if info.stdout: print (info.stdout)
+        if info.stderr: print (info.stderr)
+        # if info.stderr:
+        #     raise Exception(f'ERROR: {info.stderr}')
+        if ' step ' in info.stdout and ' f ' in info.stdout and ' grad ' in info.stdout:
+            raise Exception (f'GRADIENT ERROR')
+        if 'error' in info.stderr.lower() or info.exit_status:
+            raise Exception(f'ERROR: {info.stderr}')
+        if ' invalid number ' in info.stderr.lower():
+            raise Exception(f'ERROR: {info.stderr}')
+
+def test_setup (use_group_mulcov = False, test_asymptotic = False):
+    iota_child_true = .1
+    iota_group_true = .3 if use_group_mulcov else 0
+    iota_subgroup_true = 0.3
+    truth = dict(iota_sex_true = {'none': 0, 'g1': 0.2}, # sex effect
+                 iota_parent_true = .01,                 # value corresponding to parent with no effects
+                 iota_child_true = iota_child_true,      # absolute value of child random effects 
+                 iota_group_true = iota_group_true,      # value corresponding to group with no effects
+                 iota_subgroup_true = iota_subgroup_true) # absolute value of group random effects 
+
+    node_effects = {'p1': 0,
+                    'c1': -iota_child_true,
+                    'c2': +iota_child_true}
+
+    group_effects = {'none': truth['iota_group_true'],
+                     's1': -truth['iota_subgroup_true'],
+                     's2': +truth['iota_subgroup_true']}
+                     
+
+    # These settings affect whether or not the asymptotic statistics work
+    iota_subgroup_density = 'uniform'
+    iota_subgroup_std = None
+    if 0 and test_asymptotic:
+        iota_subgroup_density = 'gaussian'
+        iota_subgroup_std = 100
+        iota_child_density = 'gaussian'
+        iota_child_std = 100
+
+    priors = dict(parent_density = 'uniform',
+                  parent_std = None,
+                  child_density = 'uniform',
+                  child_std = None,
+                  subgroup_density = iota_subgroup_density,
+                  subgroup_std = iota_subgroup_std)
+
+    return (truth, priors, node_effects, group_effects)
+
+def run_test(file_name, test_config, truth_in,
```

Comment:
All your test names need to start with `test_` otherwise `pytest` won't see them. So this should be changed to `test_run` or something. And then the above test that doesn't actually have any asserts in it should not start with the name `test_`.

To skip running this test when you're on travis, for example, you would have:
```
def test_run(file_name, test_config, truth_in, start_from_truth, test_asymptotic, dismod):
    ...
```
It knows to skip this test in travis because of the `dismod` argument.

Reply:
create_database.py, dismod_tests.py, example_db.py are not tests. They are used by the other modules having names that start with test_   Is you comment still valid?